stacks-signer/src/runloop.rs

```diff
@@ -235,7 +236,22 @@ impl RunLoop {
                     }
                 }
             }
-            let new_signer = Signer::from(new_signer_config);
+            let mut new_signer = Signer::from(new_signer_config);
+            let dkg_id = self
+                .stacks_client
+                .get_last_round(reward_cycle)
+                .expect("Failed to get last DKG round")
+                .unwrap_or(0);
+            let approved_aggregate_key = self
+                .stacks_client
+                .get_approved_aggregate_key(reward_cycle)
+                .expect("Failed to get approved aggregate key");
```

Comment:
Fair enough.  Setting it to `None` feels quite brittle here, because the signer wouldn't be able to load the appropriate DKG shares - so we'd essentially start a broken signer unable to participate in signing rounds. But I can go the retry route like you suggested. I'll  wrap this in a retry loop, and return an error if it fails - failing the call to `self.initialize_runloop()` so that it gets retried again next pass.

Reply:
Done ✅